migrateBuckets/migrateBuckets.js

```diff
@@ -15,89 +14,90 @@ const logger = new werelogs.Logger('s3utils:migrateBuckets')
-.version('0.0.1')
+commander.version('1.0.0')
 .option('-b, --bucket-name <bucketName>', 'Name of the bucket')
-.option('-s, --src-canonical-id <srcCanonicalId',
+.option('-s, --src-canonical-id <srcCanonicalId>',
     'Canonical id of source account')
-.option('-d, --dest-canonical-id <destCanonicalId',
+.option('-d, --dest-canonical-id <destCanonicalId>',
     'Canonical id of destination account')
 .option('-o, --dest-owner-name <destOwnerName>',
     'Owner display name of destination account')
-.option('-c, --config-path <configPath>', 'Path to metadata config')
+.option('-e, --endpoint <endpoint>', 'S3Connector endpoint')
+.option('-u, --bucketd <bucketd>', 'Bootstrap list of bucketd endpoints')
+.option('-k, --https-key [httpsKey]', 'HTTPS key')
+.option('-c, --https-cert [httpsCert]', 'HTTPS cert')
+.option('-a, --https-ca [httpsCa]', 'HTTPS ca')
+.option('-p, --profile [profile]', 'AWS credentials')
 .parse(process.argv);
 
 const {
     bucketName,
     srcCanonicalId,
     destCanonicalId,
     destOwnerName,
-    configPath,
+    endpoint,
+    bucketd,
+    httpsKey,
+    httpsCert,
+    httpsCa,
+    profile,
 } = commander;
 
+if ((!ACCESS_KEY || !SECRET_KEY) && !profile) {
+    logger.error('Missing credentials. Set ACCESS_KEY & SECRET_KEY env ' +
+        'variables or pass --profile option');
+    process.exit(1);
+}
+
 if (!bucketName || !srcCanonicalId ||
-!destCanonicalId || !destOwnerName || !configPath) {
+!destCanonicalId || !destOwnerName || !bucketd) {
     logger.error('Missing command line parameter');
     commander.outputHelp();
     process.exit(1);
 }
 
-let config;
-try {
-    const data = fs.readFileSync(configPath, { encoding: 'utf-8' });
-    config = JSON.parse(data);
-} catch (err) {
-    logger.error(`Could not parse provided config file: ${err.message}`);
-    process.exit(1);
-}
+let keyInProgress;
+let versionIdInProgress;
+let nErrors = 0;
+let nObjProcessed = 0;
 
-const s3 = new AWS.S3({
-    accessKeyId: ACCESS_KEY,
-    secretAccessKey: SECRET_KEY,
+const s3Opts = {
     region: 'us-east-1',
     sslEnabled: false,
-    endpoint: ENDPOINT,
+    endpoint,
     s3ForcePathStyle: true,
     apiVersions: { s3: '2006-03-01' },
     signatureVersion: 'v4',
     signatureCache: false,
     httpOptions: {
-        maxRetries: 0,
+        maxRetries: 2,
         timeout: 0,
         agent: new http.Agent({ keepAlive: true }),
     },
-});
-
-const { bootstrap } = config.bucketd;
-if (!bootstrap || bootstrap.length === 0) {
-    logger.error('Bucketd bootstrap list is empty');
+};
+if (ACCESS_KEY && SECRET_KEY) {
+    s3Opts.accessKeyId = ACCESS_KEY;
+    s3Opts.secretAccessKey = SECRET_KEY;
 }
+if (profile) {
+    const credentials = new AWS.SharedIniFileCredentials({ profile });
+    AWS.config.credentials = credentials;
+}
+
+const s3 = new AWS.S3(s3Opts);
+
 let mdclient;
 try {
-    if (config.https) {
-        const { key, cert, ca } = config.https;
+    if (httpsKey && httpsCert && httpsCa) {
```

Comment:
I think `httpsCa` may not be always provided even if `httpsKey` and `httpsCert` are (they would use the system-installed CA then). I doubt I'd rather check that only the first two are defined to activate HTTPS.